### PR TITLE
Lambdify modified to handle functions and their derivatives simultaneously

### PR DESCRIPTION
**Pull request by arighnac**
Fixes Issue #13753 .

**NOW**

`>>>from sympy import *`
`>>>x = symbols('x')`
`>>>fun = Function('fun')(x)`
`>>>expr = fun + fun.diff()`
`>>>f = lambdify((fun,fun.diff()),expr)`
`>>>f(sin(x))`

`sin(x) + cos(x)`

The lambdify function is now modified to handle both functions and their derivatives simultaneously.

### CODE REVIEW

File: sympy/utilities/lambdify.py [hunk 1]
```diff
@@ -528,6 +528,12 @@ def sub_args(args, dummies_dict):
                 dummies = Dummy()
                 dummies_dict.update({args : dummies})
                 return str(dummies)
+            elif isinstance(args ,(Function, Derivative)):
```
Comment (asmeurer) [inline sympy/utilities/lambdify.py]: The `Function` here does nothing, as it's caught by the `if` clause above.

Comment (ArighnaIITG) [reply sympy/utilities/lambdify.py]: What do I need to do? Can you please help?
@asmeurer , @moorepants , please review.

File: sympy/utilities/tests/test_lambdify.py [hunk 2]
```diff
@@ -219,6 +220,15 @@ def test_issue_9334():
...
 
+#================== Test derivatives ===============================
+
+
+def test_derivative():
+    expr = func + func.diff()
+    f = lambdify((func, func.diff()), expr)
+    assert f(sin(x)) == sin(x) + cos(x)
+    assert f(cos(x)) == cos(x) - sin(x)
+
```
[the remark this answers is not in the document]
Comment (moorepants) [reply sympy/utilities/tests/test_lambdify.py]: Also, check whether passing in numerics to f() does the correct thing.

[the remark this answers is not in the document]
Comment (ArighnaIITG) [reply sympy/utilities/tests/test_lambdify.py]: And what do I change in the **lambdify.py** file?

**Comment by ArighnaIITG** [comment]
@moorepants , The test cases have been added, but I am stuck with the lambdify file. Please review.

**Comment by sixpearls** [comment]
Hi @ArighnaIITG, do you mind if I take this PR over?

Thanks,

**Comment by ac9888185875** [comment]
Does the following case needs to be working fine or it's not a valid case?
f = Function('f')(x)
expr = f + f.diff() + f.diff().diff()
expr2 = lambdify((f,f.diff()),expr)

**Pull request closed without merging**